lib/validators/properties-validator.ts

```diff
@@ -50,6 +51,14 @@ export class PropertiesValidator implements Validator {
                     }
                 }
             }
+
+            // Validate if the doc-media properties have a media-properties control type
+            if (compProp.dataType === 'doc-media') {
```

Comment:
This checking doesn't look correct and it is not full. The goal is to check if a component uses doc-media directive and if it does then check if the component has a property with control type media-properties and this property is applied to that directive (checking directiveKey). I was going to suggest you looking at interactive directive and property checking but it is not implemented too. At least you can take a look how property type is checked in interactive-validator, because the current one is wrong, if dataType is not doc-media then control type can be 'media-properties' - what is wrong. You have to check control type first, then dataType.

Reply:
Might indeed make sense to move it to DocMediaValidator:
- When the component has more than 1 doc-media, the component is invalid
- When it has 1 doc-media directive, it can have a property with dataType='doc-media' + control.type='media-properties'
- When it has no doc-media directive, it can't have a property with control.type='media-properties'.

Reply:
@vitaly-retel The jira task this was based on was by far not as explanatory of the required behaviour as that. I'll look into it.
@wwsno Thanks for the summary of how it should behave, I can work with this.

Reply:
PropertiesValidator checks properties list only. But everything what is related to the directives should check components. You can see that there is ComponentsDefinition and ParsedComponentsDefinition. The last one includes information about directives: which directives components have and which properties those directives use. As I said PropertiesValidator checks ComponentsDefinition (simple checking if properties list is ok). Better to place them all in DocMediaValidator. And again, the same should be done for interactive directive. Guess you can combine it somehowe because validation logic is the same for both doc-media and doc-interactive.